examples/tracingpolicy/hardlink-observe.yaml

```diff
@@ -22,4 +22,4 @@ spec:
       - index: 1
         operator: "Equal"
         values:
-        - "/etc/passwd\0"
+        - "/etc/passwd"
```

Comment:
The \0 shouldn't really be specified for equal operators as the match is now over the whole string and not a potential substring of it. That said, maybe we should handle (chop off) a \0 in a string type in case people have these in their tracing policies.

Reply:
I've added code to silently remove trailing \0 characters so it now supports (with tests) strings with and without the trailing \0.

Reply:
we have multiple examples with `\0` as well in the documentation so that's cool that it works in any case.

Reply:
Suggest after this is merged that we remove the \0 characters. It doesn't look great and could be confusing.